File: pset2/caesar.c
/**
 * caesar.c
 * 
 * Natalia Ostiak
 * Technology Nation
 * 
 * Encrypts messages using Caesar’s cipher.
 */
 
#include <stdlib.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// prototype 
void PrintLetters(char a, int b, int c);

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf(" Usage:./caesar key\n");
        return 1;
    }
    
    int k = atoi(argv[1]);
    printf("plaintext:  ");
    string s = GetString();
    printf("ciphertext: ");
    int step;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isalpha(s[i]))
        {
            if (islower(s[i]))
                step = 97;
            else 
                step = 65;
            PrintLetters(s[i], k, step);
        }
        else
            printf("%c", s[i]);
    }
    
    printf("\n");
    return 0;
}

void PrintLetters(char sf, int kf, int stepf)
{
    printf("%c", (sf - stepf + kf) % 26 + stepf);
}    

File: pset6/caesar.py
 # caesar.py
 # 
 # Natalia Ostiak
 # Technology Nation
 # 
 # Encrypts messages using Caesar’s cipher.

import sys

if len(sys.argv) != 2:
    print(" Usage:python caesar.py key")
    exit(1) 
    
k = int(sys.argv[1])
s = input("plaintext:  ")
    
print("ciphertext: ", end="")

for c in s:
    if c.isalpha():        
        if c.islower():
            step = 97
        else: 
            step = 65
        print("{}".format(chr((ord(c) - step + k)%26 + step)), end="")        
    else:
        print("{}".format(c), end="")
        
print()        
exit(0)
